src/charm.py

```diff
@@ -109,18 +101,22 @@ def filter_lines(self, line) -> bool:
 
         return timestamp > (datetime.now() - timedelta(minutes=20))
 
-    def get_ip(self, line: str) -> str:
+    def _get_ip(self, line: str) -> str:
```

Comment:
I think there is a mistake here with the type hint, if the line is en empty string, it returns `None`